### PR TITLE
Fix Rating componentWillReceiveProps update checking

### LINKED ISSUE #72: onHover is broken when passing custom react element symbols
Here is example for reproduce:
https://codesandbox.io/s/oj7mzzww36

I debug it... Problem is that `RatingAPILayer` `PureComponent` `shouldComponentUpdate` does not handle checking for react element. So, on every onHover execution, `RatingAPILayer` will be rerendered which cause multiple rerendering. Another problem is in `Rating` component `componentWillReceiveProps`:
```
  componentWillReceiveProps(nextProps) {
    this.setState({
      dirty: this.props.value !== nextProps.value && !this.state.dirty
        ? true
        : this.state.dirty,
      displayValue: nextProps.value
    });
  }
```

I do not understand why on every props update `displayValue` is set to `nextProps.value`. But if `shouldComponentUpdate` is handled properly in `RatingAPILayer` this is not a problem.

### PR DESCRIPTION
**Pull request by darkowic**
Resolves #72

### CODE REVIEW

**Comment by darkowic** [comment]
@dreyescat ping

File: src/Rating.jsx [hunk 1]
```diff
@@ -22,11 +22,12 @@ class Rating extends React.PureComponent {
   }
 
   componentWillReceiveProps(nextProps) {
+    const valueChanged = this.props.value !== nextProps.value;
     this.setState({
-      dirty: this.props.value !== nextProps.value && !this.state.dirty
+      dirty: valueChanged && !this.state.dirty
```
Comment (dreyescat) [inline src/Rating.jsx]: Not part of this pull request... But... The `dirty` state expression looks convoluted to me. Isn't it equivalent to:

```javascript
dirty: valueChanged || this.state.dirty
```

Comment (dreyescat) [reply src/Rating.jsx]: Even more... Is it used somewhere?

```bash
$ grep dirty Rating.jsx
      dirty: false
      dirty: this.props.value !== nextProps.value && !this.state.dirty
        : this.state.dirty,
```

Comment (darkowic) [reply src/Rating.jsx]: It was also confusing to me. I didn't find any usage of it.

**Pull request merged**